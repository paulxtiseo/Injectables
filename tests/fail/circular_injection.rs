use injectables::{injectable, inject_fields};

#[injectable]
#[inject_fields(B)]
pub struct A {
  pub id: u64,
}

#[injectable]
#[inject_fields(A)]
pub struct B {
  pub name: String,
}

fn main() {}