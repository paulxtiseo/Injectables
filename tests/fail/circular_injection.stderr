error: Cannot inject fields from 'B' as this type does not exist
 --> tests/fail/circular_injection.rs:4:1
  |
4 | #[inject_fields(B)]
  | ^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `inject_fields` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Circular injection chain detected: B already depends on A
  --> tests/fail/circular_injection.rs:10:1
   |
10 | #[inject_fields(A)]
   | ^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `inject_fields` (in Nightly builds, run with -Z macro-backtrace for more info)
